README.md
<h1 align="center">
  <a target="_blank" href="https://inplayer-ui.netlify.app/">
    <img src="https://assets.inplayer.com/images/inplayer-256.png" alt="inplayer-ui" title="InPlayer UI" width="300">
    <br />
    <span style="font-size: 1.5rem; color: blue">InPlayer React UI Components</span>
  </a>
</h1>
<p align="center" style="font-size: 1.2rem;">Modern React UI library. Code less, do more.</p>

[![CI]][ci]
[![Version][version-badge]][package]
[![MIT License][license-badge]][license]

```sh
yarn add @inplayer-org/inplayer-ui styled-components
```

InPlayer UI is a style system / UI library for [React](https://reactjs.org/). It works with [Styled Components 💅](https://www.styled-components.com).

## Up & Running

To install dependencies with Yarn, run:

```sh
$ yarn install
```

or to install with npm, run:

```sh
$ npm install
```

## Usage

Here is a quick example to get you started, **it's all you need**:

```jsx
import React from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";
import { ThemeProvider } from "styled-components";
import { ThemeWrapper, Button } from "@inplayer-org/inplayer-ui";

function App() {
  return (
    <ThemeWrapper withProvider={ThemeProvider}>
      <Button size="lg">
        Hello World
      </Button>
    <ThemeWrapper/>
  );
}

ReactDOM.render(<App />, document.querySelector('#app'));
```

Yes, it's really all you need to get started as you can see in this live and interactive demo:

[![Edit Button](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/k5qqq2797)

### Modifiers

This library utilizes
[`styled-components-modifiers`](https://github.com/Decisiv/styled-components-modifiers) to build
modifiers. Modifiers are small functions that allow us to alter the properties of an Element.
They primarily live in the Element's file and are solely responsible for modifying styles.
Some modifiers are common to multiple Elements. An example would be `fontWeights`.
These common modifiers live in `src/modifiers`

## Local Development

### Module Development Workflow

Helpful information on development workflow in this library lives
[here](https://gist.github.com/srdjanRakic/139502483b17c3d2d7675d5592ff8d48).

### Linting

\_**NOTE:** The linter will run against everything in the `src` directory.

### JavaScript Linting

To run the linter once:

```
$ yarn lint:js
```

To run the watch task:

```
$ yarn lint:js:watch
```

#### Style Linting

To run the style linter:

```
$ yarn lint:style
```

#### Linting JavaScript & Styles

To run both linters:

```
$ yarn lint
```

### Review

If you'd like to run the linters and tests at once (this is a nice check before pushing to
Github or deploys), you can run:

```
$ yarn review
```

### Build

\_**NOTE:** When you run `build`, Babel will create a `build` directory.

Run once:

```
$ npm/yarn run build
```

Run the watch script:

```
$ npm/yarn run build:watch
```

_**NOTE:** the build script runs in the `prepublishOnly` script just before you publish to npm._

## Contributing

We are thankful for any contributions made by the community. By contributing you agree to abide by
the Code of Conduct in the [Contributing Guidelines](https://github.com/inplayer-org/inplayer-ui/blob/master/.github/CONTRIBUTING.md).

## License

Licensed under the MIT License, Copyright © 2018-present InPlayer.

See [LICENSE](https://github.com/inplayer-org/inplayer-ui/blob/master/LICENSE) for more information.

[ci]: https://github.com/inplayer-org/inplayer-ui/workflows/CI/badge.svg
[version-badge]: https://img.shields.io/npm/v/@inplayer-org/inplayer-ui.svg?colorB=blue
[package]: https://www.npmjs.com/package/@inplayer-org/inplayer-ui
[license-badge]: https://img.shields.io/npm/l/@inplayer-org/inplayer-ui.svg?style=flat-square
[license]: https://github.com/inplayer-org/inplayer-ui/blob/master/LICENSE


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.

# [2.5.28] - 29.09.2022

## Added

- Add `disabled` for Dropdown options
# [2.5.27] - 04.08.2022

## Added

- Changed the color of the text in the footer for Previews from fixed value to a value from the theme colors

# [2.5.26] - 21.07.2022

## Added

- Add `hasItemEntries` prop for Table component

# [2.5.25] - 04.04.2022

## Added

- Add `displayFormat` prop for DayPicker component

# [2.5.24] - 09.03.2022

## Added

- New `useAnalytics` hook as an alternative to `AnalyticsComponent`

# [2.5.23] - 21.02.2022

## Change

- Buy button tag name on paywall previews

# [2.5.22] - 04.02.2022

## Added

- Option in column table to make fields(columns) editable
- onBlur function to trigger persa when input and textarea are focused-out

# [2.5.21] - 25.01.2022

## Change

- Styling changes in preview templates for restricted asset
- FooterText component refactored

# [2.5.20] - 21.01.2022

## Fixes

- TextArea by providing the required props

# [2.5.19] - 21.01.2022

## Added

- Handlers for Input, TextArea, Pagination, Accordion and Switch.
- Tagging Paywall previews

## Fixes

- TabNavigation handler type

# [2.5.18] - 10.01.2022

## Added

- Handlers for Checkbox, Radiobutton, Daypicker and Datepicker.
- Handler for selected Tab change.

# [2.5.17] - 12.11.2021

## Added

- Tagging for StepWizard

## Fixes

- Dropdown component tag sent in tracker

# [2.5.16] - 12.11.2021

## Added

- Analytics handler for dropdown component
- Tagging for pagination and navbar

# [2.5.15] - 02.11.2021

## Fixes

- Tab component import bug

### Change

- Accordion component to be analytics ready

# [2.5.14] - 27.10.2021

## Added

- Analytics handler for button component

# [2.5.13] - 05-10-2021

## Added

- AnalyticsProps type to the components in index.d.ts file.

# [2.5.12] - 28-09-2021

## Added

- Explicit exports for components, types and interfaces from `src/analytics`.

# [2.5.11] - 28-09-2021

## Added

- Analytics tracking adds forgotten import for React.

# [2.5.10] - 27-09-2021

### Added

- Analytics tracking props for all components.

# [2.5.9] - 23-09-2021

### Added

- Add another type of datepicker component with dropdown presets
- Drawer component functionality and design

# [2.5.8] - 19-08-2021

### Change

- Return Accordion panel label when toggling it.

# [2.5.7] - 17-06-2021

### Change

- Changes connected with logo re-branding

# [2.5.6] - 09-04-2021

### Fixes

- Changed width, height and padding-left of Checkbox component's top wrapper to be in `em` instead of `rem`

# [2.5.5] - 22-03-2021

### Change

- Add styling changes for asset restrictions in preview templates

# [2.5.4] - 12-03-2021

### Change

- Add missing className for Switch component in order to externaly style it with styled-components

# [2.5.3] - 23-02-2021

### Fixes

- Daypicker moment type error

# [2.5.2] - 22-01-2021

### Change

- Use pixels instead of rem for font sizes

# [2.5.1] - 15-01-2021

### Added

- Add the possibility to have more than one table button

# [2.5.0] - 12-01-2021

### Changes

- Change footer text depending on user auth state

# [2.4.2] - 11-11-2020

### Changes

- Change preview box margin for preview 1

### Fixes

- Remove polished normalize function from GlobalStyles

# [2.4.1] - 21-09-2020

### Changes

- Remove box-sizing from globalStyles

# [2.4.0] - 16-09-2020

### Changes

- Remove redundant NavBar component

# [2.3.0] - 15-09-2020

### Changes

- Remove redundant Header and UserMenu components

# [2.2.4] - 07-09-2020

### Changes

- Expose the flashTooltip method to parent component when using ref through useImperativeHandle

# [2.2.3] - 27-08-2020

### Changes

- Added shouldClose and onRequestClose optional props to Accordion
- Added actionsRowTitle optional prop to Table
- Changed Datepicker period labels

# [2.2.2] - 24-08-2020

### Fixes

- Preview 5 border styling

# [2.2.1] - 21-08-2020

### Fixes

- Downgrade babel-plugin-styled-components to version working properly with Rollup

# [2.2.0] - 20-08-2020

### Added

- Theme background color palette

# [2.1.2] - 10-08-2020

### Changes

- Added optional onLogoClick and logoTitle props to Navbar
- Added id's on TabNavigation Tabs

# [2.1.1] - 29-07-2020

### Changes

- Display next 100 years in DatePicker

# [2.1.0] - 28-07-2020

### Changes

- Custom Toggle/Switch component
- Removed icon prop from Toggle/Switch

# [2.0.4] - 20-07-2020

### Fixes

- Use Math.floor only if totalItemsNumber is 10.000 in Pagination component

# [2.0.3] - 15-07-2020

### Fixes

- Use Math.floor instead of Math.ceil in Pagination component

# [2.0.2] - 15-07-2020

### Fixes

- Set items limit to 10.000 in pagination

# [2.0.1] - 15-07-2020

### Fixes

- ESLint TypeScript errors/warnings

# [2.0.0] - 14-07-2020

### Added

- Paywall Previews Templates
- Added optional placeholder prop to DayPicker
- Select options for month and year in DatePicker and DayPicker
- Added optional disabled prop to DatePicker
- Added icon boolean prop to Switch
- Table renderEmptyTable boolean prop
- Table Header section
- Avatar component
- Custom Pagination Component
- Add more colors

### Change

- Show DatePicker's inner presets on the left
- Add optional prop to Single Day picker - disable past days
- Support one day selection on DatePicker
- Improve DatePicker to conditionally show presets
- Highlight today's date on DatePicker
- Added onClose event to SingleDay picker
- Radio button refactor and minor UI change
- Replace local images with appropriate ones from aws
- Remove react-quill and use html-react-parser
- Sanitize html string using dompurify
- Open paywall modal on preview themes image click
- Redesign of buttons and minor styling changes to Header, Card and Table
- Add ids to Paywall Previews for easier client manipulation
- UserMenu width and display
- Remove font loading
- Add disabled option to DayPicker component
- Rename InPlayer icons with more meaningful names

config/theme.js
const { rem } = require('polished');
const colors = require('./colors');

module.exports = {
  palette: {
    primary: {
      main: colors.accentBlue,
      light: colors.accentBlueFade,
      dark: colors.accentDarkBlue,
    },
    secondary: {
      main: colors.blue,
      light: colors.darkBlueFilterLight,
      dark: colors.darkBlue,
    },
    text: {
      main: colors.gray,
      light: colors.lightGray,
      disabled: colors.darkGray,
    },
  },
  dimensions: {
    baseGrid: 8,
    borderRadius: '2px',
  },
  font: {
    primary: 'Roboto',
    primaryFallback: 'Helvetica, Arial, sans-serif',
    weights: {
      light: 300,
      normal: 400,
      semiBold: 600,
      bold: 700,
    },
    sizes: {
      extraSmall: rem(12),
      small: rem(14),
      medium: rem(16),
      large: rem(18),
      h1: rem(40),
      h2: rem(36),
      h3: rem(32),
      h4: rem(28),
      h5: rem(24),
      h6: rem(20),
    },
  },
  padding: {
    sizes: {
      small: '10px 20px 11px',
      medium: '13px 22px 14px',
    },
  },
};
